src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java

```diff
@@ -75,14 +77,19 @@ public SpotifyHttpManager(Builder builder) {
       );
     }
 
-    ConnectionConfig connectionConfig = ConnectionConfig
-      .custom()
-      .setConnectTimeout(builder.connectTimeout != null
-        ? Timeout.ofMilliseconds(builder.connectTimeout)
-        : ConnectionConfig.DEFAULT.getConnectTimeout())
-      .build();
-    BasicHttpClientConnectionManager connectionManager = new BasicHttpClientConnectionManager();
-    connectionManager.setConnectionConfig(connectionConfig);
+    if (builder.connectionManager == null) {
+      BasicHttpClientConnectionManager basicHttpClientConnectionManager = new BasicHttpClientConnectionManager();
+      basicHttpClientConnectionManager.setConnectionConfig(ConnectionConfig
+        .custom()
+        .setConnectTimeout(connectTimeout != null ?
+          Timeout.ofMilliseconds(connectTimeout) :
+          ConnectionConfig.DEFAULT.getConnectTimeout())
+        .build());
+      this.connectionManager = basicHttpClientConnectionManager;
+    } else {
+      this.connectionManager = builder.connectionManager;
+    }
```

Comment:
nitpick consider conditional consistency throughout the library.

either this:
```
    if (value == null) {
      // something
    } else {
      // something else
    }
```

or this:
```
    if (value != null) {
      // something
    } else {
      // something else
    }
```

Reply:
Switched :white_check_mark: